packages/cli/src/lib/push/addCommon.ts

```diff
@@ -0,0 +1,94 @@
+import {CliUx} from '@oclif/core';
+import {red, dim, green} from 'chalk';
+import {BuiltInTransformers, errors} from '@coveo/push-api-client';
+import {normalizeInvalidFields} from '../flags/sourceCommonFlags';
+import {pluralizeIfNeeded} from '../utils/string';
+import dedent from 'ts-dedent';
+
+const allowedCharRegExp = new RegExp('^[a-z]+[a-z0-9_]*$');
+interface PrintableUnsupportedField {
+  original: string;
+  normalized: string;
+  valid: boolean;
+}
+
+export const formatErrorMessage = (err: unknown) => {
+  const {UnsupportedFieldError} = errors;
+  if (err instanceof UnsupportedFieldError) {
+    const normalizations = getNormalizations(err.unsupportedFields);
+    const fixables = normalizations.flatMap(colorizeFixableField);
+    const unfixables = normalizations.flatMap(colorizeUnFixableField);
+    const isFixable = unfixables.length === 0;
+
+    printInvalidFieldTable(fixables);
+    printInvalidFieldTable(unfixables);
+    err.message = getFormattedMessage(isFixable);
+  }
+};
+
+const getFormattedMessage = (isFixable: boolean) => {
+  const normalizeFlag = Object.keys(normalizeInvalidFields())[0];
+  let message = `Run the same command using the \`--${normalizeFlag}\` flag to automatically normalize field names while pushing your data.`;
+  if (!isFixable) {
+    message = dedent`Cannot normalize some of the invalid field names detected in your data
+      Review your data and ensure all field names as shown in the second table have at least one alphabetic character.`;
+  }
+  return message;
+};
+
+const colorizeFixableField = (
+  field: PrintableUnsupportedField
+): PrintableUnsupportedField | [] => {
```

Comment:
I think you're not using flatmap properly, you're supposed to return an array at each steps iirc.
> Return a 1-element array to keep the item, a multiple-element array to add items, or a 0-element array to remove the item.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#for_adding_and_removing_items_during_a_map

Reply:
No. That's the beauty of it. The callback can either return a element of the final array as the element itself or in an array
![image](https://user-images.githubusercontent.com/12199712/170505151-6bcb607e-b743-4230-9e82-627691024aa0.png)